developer/src/kmc-kmn/src/compiler/compiler.ts

```diff
@@ -551,12 +551,18 @@ export class KmnCompiler implements KeymanCompiler, UnicodeSetParser {
       return null;
     }
 
-    // TODO-LDML: Catch OOM
+    if ((rangeCount * 2) < 0) {
+      throw new RangeError(`Internal error: negative rangeCount * 2 = ${rangeCount * 2}`);
+    }
     const buf = this.wasmExports.malloc(rangeCount * 2 * Module.HEAPU32.BYTES_PER_ELEMENT);
+    if (buf <= 0) {
+      // out of memory will return zero.
+      throw new RangeError(`Internal error: wasm malloc() returned ${buf}`);
+    }
     // fix \u1234 pattern format
     pattern = KmnCompiler.fixNewPattern(pattern);
-    /** If <= 0: return code. If positive: range count */
     const rc = Module.kmcmp_parseUnicodeSet(pattern, buf, rangeCount * 2);
+    /** If <= 0: error return code. If positive: it's a range count */
     if (rc >= 0) {
```

Comment:
The comment and the if statement don't match up. Should it be < 0 for error or <= 0 for error?

Reply:
Yes, thank you.  Fixed the comment, and clarified why zero is a valid (non-error) return.
Appreciate the review.